src/components/Charts/Axes/Axes.js

```diff
@@ -1,28 +1,34 @@
 import React from 'react'
 import _get from 'lodash.get'
 import { v4 as uuid } from 'uuid'
-
 import { Axis } from '../Axis'
 import AxisSelector from './AxisSelector'
+import RowNumberSelector from './RowNumberSelector'
 import { getBBoxFromRef } from '../../../js/Util'
 import { axesDefaultProps, axesPropTypes } from '../helpers'
 
 export default class Axes extends React.Component {
   constructor(props) {
     super(props)
-
+    this.COMPONENT_ID = uuid()
     this.xAxisKey = uuid()
     this.yAxisKey = uuid()
     this.axisLabelPaddingTop = 5
     this.axisLabelPaddingLeft = 10
-
+    this.maxRows = 5000
+    this.initialRowNumber = 50
+    this.visualizingSpan = document.getElementById('visualizing-span')
```

Comment:
Just to confirm, this constructor is only called once before it mounts. If visualizing-span is rendered in this component, then it won't exist at this time.

Also we probably want to add a unique identifier to the ID `visualizing-span`. If you are in a dashboard it will take the first element it finds, which could be in a different QueryOutput